internal/controller/supervisorstorage/garbage_collector.go

```diff
@@ -121,28 +121,40 @@ func (c *garbageCollectorController) Sync(ctx controllerlib.Context) error {
 			continue
 		}
 
-		if garbageCollectAfterTime.Before(frozenClock.Now()) {
-			storageType, isSessionStorage := secret.Labels[crud.SecretLabelKey]
-			if isSessionStorage {
-				err := c.maybeRevokeUpstreamOIDCRefreshToken(ctx.Context, storageType, secret)
-				if err != nil {
-					// Log the error for debugging purposes, but still carry on to delete the Secret despite the error.
-					plog.WarningErr("garbage collector could not revoke upstream refresh token", err, logKV(secret))
-				}
-			}
+		if !garbageCollectAfterTime.Before(frozenClock.Now()) {
+			// not old enough yet
+			continue
+		}
 
-			err = c.kubeClient.CoreV1().Secrets(secret.Namespace).Delete(ctx.Context, secret.Name, metav1.DeleteOptions{
-				Preconditions: &metav1.Preconditions{
-					UID:             &secret.UID,
-					ResourceVersion: &secret.ResourceVersion,
-				},
-			})
+		storageType, isSessionStorage := secret.Labels[crud.SecretLabelKey]
+		if isSessionStorage {
+			err := c.maybeRevokeUpstreamOIDCRefreshToken(ctx.Context, storageType, secret)
 			if err != nil {
-				plog.WarningErr("failed to garbage collect resource", err, logKV(secret))
-				continue
+				// Log the error for debugging purposes, but still carry on to delete the Secret despite the error.
+				plog.WarningErr("garbage collector could not revoke upstream refresh token", err, logKV(secret))
```

Comment:
Oh good catch. That's an old mistake. I don't think we noticed this because it was printing all of the information to the log before, its just that the format was slightly off. E.g. old log:

`pinniped-supervisor I1123 19:49:35.963129       1 garbage_collector.go:156] "storage garbage collector deleted resource" [secretName pinniped-storage-refresh-token-75ju2dvrwuq6ljyx7yxto7gf3bkino74e3a4nkwesckgkj3wv25q secretNamespace supervisor secretType storage.pinniped.dev/refresh-token garbageCollectAfter 2021-11-23T19:47:39Z storageTypeLabelValue refresh-token]="(MISSING)"`

After fixing, the format is improved as expected:

`pinniped-supervisor I1123 19:57:59.462157       1 garbage_collector.go:156] "storage garbage collector deleted resource" secretName="pinniped-storage-refresh-token-od76mvb2crxchd6mlhn2tvezhj3j336u7zsvzyi6u6jwgeoi2ucq" secretNamespace="supervisor" secretType="storage.pinniped.dev/refresh-token" garbageCollectAfter="2021-11-23T19:56:43Z" storageTypeLabelValue="refresh-token"`